tests/integration/pure_cairo_vm/contracts/roll_warp_tester.cairo

```diff
@@ -3,49 +3,71 @@
 from starkware.cairo.common.cairo_builtins import HashBuiltin
 from starkware.starknet.common.syscalls import get_block_timestamp, get_block_number
 
+@storage_var
+func stored_block_timestamp() -> (res: felt) {
+}
+
+@storage_var
+func stored_block_number() -> (res: felt) {
+}
+
 @view
-func timestamp_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
-    let (block_timestamp) = get_block_timestamp();
+func stored_block_timestamp_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+    let (block_timestamp) = stored_block_timestamp.read();
     return (res=block_timestamp);
 }
 
 @view
-func block_number_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
-    let (block_number) = get_block_number();
+func stored_block_number_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+    let (block_number) = stored_block_number.read();
     return (res=block_number);
 }
 
-@storage_var
-func stored_block_timestamp() -> (res: felt) {
+@view
+func timestamp_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+    let (block_timestamp) = get_block_timestamp();
+    return (res=block_timestamp);
 }
 
-@storage_var
-func stored_block_number() -> (res: felt) {
+@view
+func block_number_getter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+    let (block_number) = get_block_number();
+    return (res=block_number);
 }
 
-@view
-func block_timestamp_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+@external
+func block_timestamp_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
     let (block_timestamp) = get_block_timestamp();
     stored_block_timestamp.write(block_timestamp);
 
-    return (res=block_timestamp);
+    return ();
 }
 
-@view
-func block_number_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
+@external
+func block_number_setter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() {
     let (block_number) = get_block_number();
     stored_block_number.write(block_number);
-    return (res=block_number);
+
+    return ();
 }
```

Comment:
I think we should change the naming of these functions.

The naming is really confusing to me and the names don't really describe what they do exactly. I think we should change them similarly to the bindings - to indicate actions, not the roles (and do it in a slightly better/more-readable manner).

| current name | what it does | proposed new name | 
| ------------- | ------------- | ------------- | 
| stored_block_timestamp_getter | returns stored timestamp | get_stored_block_timestamp |
| stored_block_number_getter | returns stored block number | get_stored_block_number |
| timestamp_getter | returns a timestamp using a syscall  | get_block_timestamp_wrapper |
| block_number_getter | returns block number using a syscall | get_block_number_wrapper |
| block_timestamp_setter | sets the stored timestamp to the one from the syscall | set_stored_timestamp_to_syscall_value |
| block_number_setter | sets the stored block number to the one from the syscall | set_block_number_to_syscall_value |

If you don't like any of the proposed names, that's ok, but please consider making these functions' names more descriptive.

Reply:
Agree, will do.